### DISCUSSION TITLE
Are wildcard subdomains supported?

### DISCUSSION THREAD

**Discussion by alerque** [kpcyrd/acme-redirect#24] [Q&A]
If so, how to do so should probably be documented.

If not, can they be?

My use case is trying to setup an Apache HTTPS proxy in front of a GitLab Pages daemon. The latter has it's own certificate handling built in but using it requires running it on both port 80 and 443. That of course is precluded by my having only a single IP to work with, running acme-redirect on port 80 and Apache on 443 (with a mix of native sites and proxies to other daemons).

If it matters, by hope would be to get a wildcard cert for sub-sub domains (the main domain for my pages instance being itself a subdomain). If that's not possible of course a regular subdomain wildcard could work, I'd just have to purpose a new domain for it.

**Comment by kpcyrd** [comment]
Unfortunately wildcard domains are currently only available with the [dns-01 challenge](https://letsencrypt.org/docs/challenge-types/#dns-01-challenge), while acme-redirect only implements http-01.

You'd either need an acme implementation that is able to use api keys for your dns provider (if your provider has an api), some kind of certificate distribution system that is able to copy files between your web- and your dns server, or you could delegate the `_acme-challenge` subdomain to your webserver, which then needs to run some kind of stub dns server.

The last option could be implemented in acme-redirect, but that's likely not something I'm going to work on in the foreseeable future.
